### PR TITLE
feat: added support for encrypting events persisted in sqlite3 db

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

added support for encrypting events persisted in sqlite3 db

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [x] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/example/ios/Podfile.lock [hunk 1]
```diff
@@ -101,13 +101,16 @@ PODS:
   - GoogleUtilities/UserDefaults (7.11.5):
     - GoogleUtilities/Logger
   - Leanplum-iOS-SDK (4.0.0)
+  - MetricsReporter (1.0.0):
```
Comment (bardisg) [inline packages/example/ios/Podfile.lock:104]: do we also add metrics in flutter?

File: packages/plugins/rudder_plugin_interface/lib/src/models/rudder_mobile_config.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  bool get trackLifecycleEvents => _trackLifecycleEvents;
+
+  int get sleepTimeOut => _sleepTimeOut;
+
+  int get configRefreshInterval => _configRefreshInterval;
+
+  DBEncryption? get dbEncryption => _dbEncryption;
+}
+
+class DBEncryption {
```
Comment (bardisg) [inline packages/plugins/rudder_plugin_interface/lib/src/models/rudder_mobile_config.dart:57]: define one class per file if possible. can we split DBEncryption on its own file?

File: packages/example/ios/Podfile.lock (hunk 1 above)
Comment (desusai7) [reply packages/example/ios/Podfile.lock:104]: no we haven't added support for it in the Flutter SDK, but the `MetricsReporter` pod got downloaded, as it was a dependency of the `Rudder` iOS pod

File: packages/plugins/rudder_plugin_interface/lib/src/models/rudder_mobile_config.dart (hunk 2 above)
Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/lib/src/models/rudder_mobile_config.dart:57]: yes, will do it

Comment (desusai7) [reply packages/plugins/rudder_plugin_interface/lib/src/models/rudder_mobile_config.dart]: pushed the change

**Review by bardisg — APPROVED** [review]

**Pull request merged**